### DISCUSSION TITLE
Customize server port setting

### DISCUSSION THREAD

**Discussion by AdjWang** [rustdesk/rustdesk#1450] [Q&A]
I'm using docker and it works fine with default port setting.
```
docker image pull rustdesk/rustdesk-server
docker run --name hbbs -p 21115:21115 -p 21116:21116 -p 21116:21116/udp -p 21118:21118 -v `pwd`:/root -td --net=host rustdesk/rustdesk-server hbbs
docker run --name hbbr -p 21117:21117 -p 21119:21119 -v `pwd`:/root -td --net=host rustdesk/rustdesk-server hbbr 
```
![screenshot_20220905190459](https://user-images.githubusercontent.com/55283621/188434751-e770269a-d02e-4420-b3b4-553ea1f1942c.png)

But how to set the 3 server addresses if I change the ports? For example
```
docker image pull rustdesk/rustdesk-server
docker run --name hbbs -p 8005:21115 -p 8006:21116 -p 8006:21116/udp -p 8008:21118 -v `pwd`:/root -td --net=host rustdesk/rustdesk-server hbbs
docker run --name hbbr -p 8007:21117 -p 8009:21119 -v `pwd`:/root -td --net=host rustdesk/rustdesk-server hbbr 
```
I tried some combinations but none of them works. It only shows state: `Connecting to the RustDesk network...`
I have tried `ID server: ****:8006` and `ID server: ****:8006 Replay server: ****:8007`, and tested the tcp and udp connection on 800*  ports.

**Comment by akiooo45** [comment]
21116:21116    left is server prot:right is contaiter port.
You just edit left port

**Comment by AdjWang** [comment]
Thanks for your reply anyway! @li-dianzheng 
After struggling some days, I find out it was the `--net=host` disabled the port mapping of docker. The document of docker said it [here](https://docs.docker.com/network/host/), and it also warned `WARNING: Published ports are discarded when using host network mode` at container starting. I was just too blind to see it. My mistake.
I'm now using: (remove `--net=host`)
```
docker run --name hbbs -p 8005:21115 -p 8006:21116 -p 8006:21116/udp -p 8008:21118 -v `pwd`:/root -td rustdesk/rustdesk-server hbbs
docker run --name hbbr -p 8007:21117 -p 8009:21119 -v `pwd`:/root -td rustdesk/rustdesk-server hbbr 
```
and only fills `ID server: ****:8006` on the RustDesk client, it works fine.

Also, [the RustDesk documents](https://github.com/rustdesk/doc.rustdesk.com/tree/master/content/self-host/install) demonstrates the docker example using `-p` and `--net=host` simultaneously. It works but incurs a warning `WARNING: Published ports are discarded when using host network mode`. Maybe turn this to an issue? I mean not to use `--net=host` as default.

**Reply by akiooo45** [reply]
If you already have the port mapped, you don't need to add it `--net=host`
